### PR TITLE
feat: add swagger api doc

### PR DESCRIPTION
**Pull request by Abirdcfly**
#### What type of PR is this?
/kind documentation
/kind feature

#### What this PR does / why we need it

graphql comes with its own documentation, and the restful part should also come with documentation in the code, so that we can make sure that the documentation is not out of date while updating the code.（I hope so)

graphql doc show in its playground like this:
CleanShot 2023-12-25 at 17 12 48

#### Which issue(s) this PR fixes

Fixes #

#### Special notes for your reviewer

you can review the swagger doc in https://petstore.swagger.io/, just use swagger.json path: `https://raw.githubusercontent.com/Abirdcfly/arcadia/log/apiserver/docs/swagger.json`
CleanShot 2023-12-25 at 16 16 34

default path is `host:port/swagger/index.html`

### CODE REVIEW

File: apiserver/pkg/chat/chat_type.go [hunk 1]
```diff
@@ -27,56 +27,77 @@ import (
...
+	// AppNamespace, the namespace of the application
+	AppNamespace string `json:"app_namespace" binding:"required" example:"arcadia"`
 }
 
 type ConversionReqBody struct {
-	APPMetadata  `json:",inline"`
-	ConversionID string `json:"conversion_id"`
+	APPMetadata `json:",inline"`
+	// ConversionID, if it is empty, a new conversation will be created
+	ConversionID string `json:"conversion_id" exmaple:"5a41f3ca-763b-41ec-91c3-4bbbb00736d0"`
```
Comment (bjwswang) [inline apiserver/pkg/chat/chat_type.go:47]: ConversionID -> `ConversationID`

File: apiserver/pkg/chat/chat_type.go [hunk 2]
```diff
@@ -27,56 +27,77 @@ import (
...
 	Debug             bool `json:"-"`
 }
 
 type ChatRespBody struct {
-	ConversionID string                `json:"conversion_id"`
-	MessageID    string                `json:"message_id"`
-	Message      string                `json:"message"`
-	CreatedAt    time.Time             `json:"created_at"`
-	References   []retriever.Reference `json:"references,omitempty"`
+	ConversionID string `json:"conversion_id" example:"5a41f3ca-763b-41ec-91c3-4bbbb00736d0"`
```
Comment (bjwswang) [inline apiserver/pkg/chat/chat_type.go:68]: same as above

File: apiserver/pkg/chat/chat_type.go [hunk 3]
```diff
@@ -27,56 +27,77 @@ import (
...
+	MessageID    string `json:"message_id" example:"4f3546dd-5404-4bf8-a3bc-4fa3f9a7ba24"`
+	// Message is what AI say
+	Message string `json:"message" example:"旷工最小计算单位为0.5天。"`
+	// CreatedAt is the time when the message is created
+	CreatedAt time.Time `json:"created_at" example:"2023-12-21T10:21:06.389359092+08:00"`
+	// References is the list of references
+	References []retriever.Reference `json:"references,omitempty"`
 }
 
 type Conversion struct {
```
Comment (bjwswang) [inline apiserver/pkg/chat/chat_type.go:78]: same as above

File: apiserver/service/router.go [hunk 4]
```diff
@@ -49,12 +52,13 @@ func NewServerAndRun(conf config.ServerConfig) {
...
 	}
-
 	bffGroup := r.Group("/bff")
 	chatGroup := r.Group("/chat")
 	RegisterMinIOAPI(bffGroup, conf)
 	RegisterGraphQL(r, bffGroup, conf)
-	RegisterChat(chatGroup, conf)
+	registerChat(chatGroup, conf)
+	docs.SwaggerInfo.BasePath = "/"
+	r.GET("swagger/*any", ginSwagger.WrapHandler(swaggerfiles.Handler))
```
Comment (bjwswang) [inline apiserver/service/router.go:61]: Should we add a argument to controll whether or not to  expose this swagger api?

File: go.mod [hunk 5]
```diff
@@ -162,4 +167,4 @@ require (
 
 replace github.com/amikos-tech/chroma-go => github.com/bjwswang/chroma-go v0.0.0-20231011091545-0041221c9bb3
```
Comment (bjwswang) [inline go.mod:168]: Can you remove this line if this has been fixed?

**Comment by Abirdcfly** [comment]
@zggmd, please note, this pull request change one typo in chat api, from `conversion` to `conversation`, https://github.com/kubeagi/portal also need update.

File: apiserver/service/router.go [hunk 6]
```diff
@@ -49,12 +52,15 @@ func NewServerAndRun(conf config.ServerConfig) {
...
 		oidc.InitOIDCArgs(conf.IssuerURL, conf.MasterURL, conf.ClientSecret, conf.ClientID)
 	}
-
 	bffGroup := r.Group("/bff")
 	chatGroup := r.Group("/chat")
 	RegisterMinIOAPI(bffGroup, conf)
 	RegisterGraphQL(r, bffGroup, conf)
-	RegisterChat(chatGroup, conf)
+	registerChat(chatGroup, conf)
+	if conf.EnablePlayground {
```
Comment (nkwangleiGIT) [inline apiserver/service/router.go:60]: should this be EnableSwagger?

**Pull request merged**